server/pbench/bin/index-pbench

```diff
@@ -686,116 +1060,71 @@ def mk_sosreports(tb):
+    d = {}
+    try:
+        d['pbench-rpm-version'] = mdconf.get("pbench", "rpm-version")
+    except Exception:
+        d['pbench-rpm-version'] = "Unknown"
+    return d
 
-###########################################################################
-# mk-toc-source
+def _to_utc(l, u):
+    """l is a timestamp in local time and u is one in UTC. These are
+    supposed to be close, so we calculate an offset rounded to a
+    half-hour and add it to the local date. We then convert back to an
+    ISO format date.
+    """
+    lts = datetime.strptime(l, "%Y-%m-%dT%H:%M:%S")
+    uts = datetime.strptime(u[:u.rfind('.')] , "%Y-%m-%dT%H:%M:%S")
+    offset = lts - uts
+    res = round(float(offset.seconds) / 60 / 60 + offset.days * 24, 1)
+    return (lts - timedelta(0, int(res*60*60), 0)).isoformat()
 
-def mk_toc(tb, md5sum):
-    members = tb.getmembers()
-    prefix = members[0].name.split(os.path.sep)[0]
-    # create a list of dicts where each directory that contains files is represented
-    # by a dictionary with a 'name' key, whose value is the pathname of the dictionary
-    # and a 'files' key, which is a list of the files contained in the directory.
-    if options.debug_unittest:
-        # sort the list to produce a predictable result for comparison to gold output
-        import collections
-        d = collections.OrderedDict()
-    else:
+def mk_run_metadata(mdconf, md5sum, dirname, dirprefix):
+    try:
         d = {}
-    for m in members:
-        if m.isdir():
-            dname = os.path.dirname(m.name)[len(prefix):] + os.path.sep
-            if dname not in d:
-                d[dname] = { 'directory': dname }
-        elif m.isfile() or m.issym():
-            dname = os.path.dirname(m.name)[len(prefix):] + os.path.sep
-            fentry = { 'name' : os.path.basename(m.name),
-                       'size' : m.size,
-                       'mode' : m.mode
-            }
-            if m.issym():
-                fentry['linkpath'] = m.linkpath
-            if dname in d:
-                # we may have created an empty dir entry already
-                # without a 'files' entry, so we now make sure that
-                # there is an empty 'files' entry that we can populate.
-                if 'files' not in d[dname]:
-                    d[dname]['files'] = []
-                d[dname]['files'].append(fentry)
-            else:
-                # presumably we'll never fall through here any more:
-                # the directory entry always preceded the file entry
-                # in the tarball.
-                d[dname] = { 'directory': dname, 'files': [fentry] }
-        else:
-            # if debug: print(repr(m))
-            continue
-    return (prefix, d)
+        d.update(mdconf.items('run'))
+        d.update(mdconf.items('pbench'))
+        # Fix up old format dates so ES won't barf on them.
+        d['start_run'] = d['start_run'].replace('_', 'T')
+        d['end_run'] = d['end_run'].replace('_', 'T')
```

Comment:
We have to handle this replacement of "`_`" with "`T`" for legacy reasons, but we should also see about fixing this in the agent `metadata.log` generation.  @ndokos or @chaitanyaenr, can one of you look at this?